src/inc/sift-events/sift-events.php

```diff
@@ -564,102 +563,168 @@ public static function create_order( string $order_id, \WC_Order $order ) {
+			// Log when event can't be sent due to settings.
+			if ( function_exists( 'wc_get_logger' ) ) {
+				wc_get_logger()->debug(
+					sprintf( 'Event %s disabled for order %s', $event, $order_id ),
+					array( 'source' => 'sift-order-events' )
+				);
+			}
 			return;
 		}
 
+		$sift_order = Sift_For_WooCommerce::get_sift_order_from_wc_order( $order );
+
 		// Determine user and session context.
-		$user_id  = wp_get_current_user()->ID ?? null; // Check first for logged-in user.
-		$is_admin = 1 === $user_id;
+		$user_id    = wp_get_current_user()->ID ?? null; // Check first for logged-in user.
+		$user_id    = self::format_user_id( intval( $user_id ) );
+		$user_email = wp_get_current_user()->user_email ?? $order->get_billing_email() ?? null;
 
-		// Figure out if it should use the session ID if no logged-in user exists.
-		if ( ! $user_id || $is_admin ) {
-			$user_id = $order->get_user_id() ?? null; // Use order user ID if it isn't available otherwise
+		// If there is no user ID, fall back to the billing email as the user ID.
+		if ( ! $user_id ) {
+			$user_id = $user_email;
 		}
 
-		$physical_or_electronic = '$electronic';
-		$items                  = array();
-		foreach ( $order->get_items( 'line_item' ) as $item ) {
-			if ( ! $item instanceof WC_Order_Item_Product ) {
-				// log an error...
-				wc_get_logger()->error( sprintf( 'Item not Item Product (order: %d).', $order->get_id() ) );
-				continue;
-			}
-			// Most of this we're basing off return value from `WC_Order_Item_Product::get_product()` as it will return the correct variation.
+		$browser = self::get_client_browser();
+		$ip      = $order->get_customer_ip_address() ?? self::get_client_ip();
+
+		$properties = array(
+			'$user_id'            => $user_id,
+			'$session_id'         => WC()->session?->get_customer_unique_id() ?? '',
+			'$order_id'           => $order_id,
+			'$user_email'         => $user_email,
+			'$verification_phone_number'
+				=> '+' === substr( $order->get_billing_phone(), 0, 1 ) ? preg_replace( '/[^0-9\+]/', '', $order->get_billing_phone() ) : null,
+			'$amount'             => self::get_transaction_micros( floatval( $order->get_total() ) ),
+			'$payment_methods'    => $sift_order->get_payment_methods(),
+			'$currency_code'      => $order->get_currency(),
+			'$billing_address'    => self::get_order_address( $order_id, 'billing' ),
+			'$shipping_address'   => self::get_order_address( $order_id, 'shipping' ),
+			'$expedited_shipping' => false,
+			'$items'              => array(),
+			'$browser'            => $browser,
+			'$ip'                 => $ip,
+			'$time'               => intval( 1000 * microtime( true ) ),
+		);
+
+		// Add the meta data.
+		foreach ( $order->get_items() as $item ) {
 			$product = $item->get_product();
-			if ( empty( $product ) ) {
-				// log an error...
-				wc_get_logger()->error( sprintf( 'Product not found for order %d.', $order->get_id() ) );
+
+			wc_get_logger()->debug(
```

Comment:
And here you haven't checked for `wc_get_logger`?

Reply:
Added a new log() method to handle these.